### PR TITLE
Fix for 323

### PR DESCRIPTION
**Pull request by hweeks**
### Summary:

This should fix the redeclaration of options when creating a new model with an already defined schema.

### Code sample:

There is a live code sample available in `/issue_reproduction/323/model.js`, please use this as a bar for correctness

### GitHub linked issue:

#323 

### Is this ready to be merged into Dynamoose? (select 1):
- [x] Yes
- [ ] No

### Other:
- [x] I have searched through the GitHub pull requests to ensure this PR has not already been submitted
- [ ] I have updated the Dynamoose documentation (if required) given the changes I made
- [x] I have added/updated the Dynamoose test cases (if required) given the changes I made
- [x] I agree that all changes made in this pull request may be distributed and are made available in accordance with the [Dynamoose license](https://github.com/dynamoosejs/dynamoose/blob/master/LICENSE.txt)
- [x] All of my commits and commit messages are detailed, explain what changes were made, and are easy to follow and understand
- [x] I have filled out all fields above

### CODE REVIEW

File: test/Model.js [hunk 1]
```diff
@@ -169,6 +169,20 @@ describe('Model', function (){
...
+  it('Should support schema option updates via constructor', async function () {
+    this.timeout(12000);
+
+    const Wolf3 = dynamoose.model('Wolf32', new dynamoose.Schema({
+      id: Number,
+      name: {
+        type: String,
+      },
+      friends: Map,
+    }), {saveUnknown: ['friends'],});
```
Comment (hweeks) [inline test/Model.js]: this is the key part, the model call has a temp override, and needs to be allowed

File: package.json [hunk 2]
```diff
@@ -63,12 +63,14 @@
...
     "coveralls": "3.0.2",
     "cz-conventional-changelog": "2.1.0",
+    "dynalite": "^2.2.0",
     "dynamodb-local": "0.0.24",
     "eslint": "5.13.0",
     "husky": "1.3.1",
     "mocha": "5.2.0",
     "nyc": "13.1.0",
     "semantic-release": "15.13.3",
+    "shortid": "^2.2.14",
```
Comment (hweeks) [inline package.json]: i'd like to leave these two here, i believe there is a case for adding the id functionality to dynamoose, and honestly dynalite might be a nice change of pace for test speed

**Comment by fishcharlie** [comment]
@hweeks Merge conflicts need to be resolved

**Comment by fishcharlie** [comment]
@hweeks Let me know your thoughts. I think we should close this PR without merging.

**Pull request closed without merging**